api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -278,7 +278,7 @@ public Obs voidObs(Obs obs, String reason) throws APIException {
 	 * @throws APIException
 	 */
 	public Obs unvoidObs(Obs obs) throws APIException {
-		return dao.saveObs(obs);
+		return Context.getObsService().saveObs(obs,"unvoid obs");
```

Comment:
What I want to know is the behavior for the scenarios below after your changes;

- One voids and unvoid an obs with no changes.
- One voids an obs with changes and then unvoids it, there is a ticket about discarding any changes when voiding an item which would partly address this particular scenario.
- One voids an obs with no changes and unvoids it with changes.

I guess the question is what happens to the unvoided obs and is there any new Obs getting created to repalce it?